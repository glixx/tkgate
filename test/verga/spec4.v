module top;
  integer i;
  reg ci = 1'b0;
  reg [1:0] a,b;
  wire [1:0] s;
  wire co;

  initial
    $monitor("%4d: a=%d b=%d ci=%d - s=%d co=%d",$time,a,b,ci,s,co);

  initial
    for (i = 0;i < 16;i = i + 1)
      begin
	{a,b} = i;
	#100;
      end

  _GGADD8 f(a,b,s,ci,co);

endmodule

module _GGADD8 (A, B, S, CI, CO);
input CI;
output CO;
input [7:0] A,B;
output [7:0] S;

  specify
    (A,B *> S) = 1;
    (A,B *> CO) = 1;
    (CI *> S) = 1;
    (CI *> CO) = 1;
  endspecify

  assign {CO,S} = A + B + CI;

endmodule
