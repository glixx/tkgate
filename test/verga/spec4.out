Warning: <"spec4.v", 18> Port size mismatch on input port 'A'.
Warning: <"spec4.v", 18> Port size mismatch on input port 'B'.
Warning: <"spec4.v", 18> Port size mismatch on output port 'S'.
   0: a=0 b=0 ci=0 - s=x co=x
   1: a=0 b=0 ci=0 - s=0 co=0
 100: a=0 b=1 ci=0 - s=0 co=0
 101: a=0 b=1 ci=0 - s=1 co=0
 200: a=0 b=2 ci=0 - s=1 co=0
 201: a=0 b=2 ci=0 - s=2 co=0
 300: a=0 b=3 ci=0 - s=2 co=0
 301: a=0 b=3 ci=0 - s=3 co=0
 400: a=1 b=0 ci=0 - s=3 co=0
 401: a=1 b=0 ci=0 - s=1 co=0
 500: a=1 b=1 ci=0 - s=1 co=0
 501: a=1 b=1 ci=0 - s=2 co=0
 600: a=1 b=2 ci=0 - s=2 co=0
 601: a=1 b=2 ci=0 - s=3 co=0
 700: a=1 b=3 ci=0 - s=3 co=0
 701: a=1 b=3 ci=0 - s=0 co=0
 800: a=2 b=0 ci=0 - s=0 co=0
 801: a=2 b=0 ci=0 - s=2 co=0
 900: a=2 b=1 ci=0 - s=2 co=0
 901: a=2 b=1 ci=0 - s=3 co=0
1000: a=2 b=2 ci=0 - s=3 co=0
1001: a=2 b=2 ci=0 - s=0 co=0
1100: a=2 b=3 ci=0 - s=0 co=0
1101: a=2 b=3 ci=0 - s=1 co=0
1200: a=3 b=0 ci=0 - s=1 co=0
1201: a=3 b=0 ci=0 - s=3 co=0
1300: a=3 b=1 ci=0 - s=3 co=0
1301: a=3 b=1 ci=0 - s=0 co=0
1400: a=3 b=2 ci=0 - s=0 co=0
1401: a=3 b=2 ci=0 - s=1 co=0
1500: a=3 b=3 ci=0 - s=1 co=0
1501: a=3 b=3 ci=0 - s=2 co=0
